### PR TITLE
Add TS Bindings

### PR DESCRIPTION
**Pull request by piraka9011**
Signed-off-by: Anas Abou Allaban <aabouallaban@pm.me>

### CODE REVIEW

File: native_client/javascript/Makefile [hunk 1]
```diff
@@ -42,7 +42,8 @@ node-wrapper: copy-deps build
 	$(NODE_BUILD_TOOL) $(NODE_PLATFORM_TARGET) $(NODE_RUNTIME) $(NODE_ABI_TARGET) $(NODE_DIST_URL) package $(NODE_BUILD_VERBOSE)
 
 npm-pack: clean package.json index.js
-	npm install node-pre-gyp@0.14.x
+	npm install node-pre-gyp@0.14.x dts-gen@0.6.0
+	dts-gen -module deepspeech -file index.d.ts
```
Comment (lissyx) [inline native_client/javascript/Makefile]: So, I got tricked and `dts-gen` is not in the `PATH`, though, once I call it with proper path on my local machine, it fails in weird way:
```
make: Entering directory '/home/alex/codaz/Mozilla/DeepSpeech/deepspeech-kdavis/native_client/javascript'
rm -f deepspeech_wrap.cxx package.json
rm -rf ./build/
sed \
	-e 's/$(PROJECT_NAME)/deepspeech/' \
	-e 's/$(PROJECT_VERSION)/0.7.0-alpha.3/' \
	package.json.in > package.json && cat package.json
{
    "name" : "deepspeech",
    "version" : "0.7.0-alpha.3",
    "description" : "DeepSpeech NodeJS bindings",
    "main" : "./index",
    "bin": {
       "deepspeech": "./client.js"
    },
    "author" : "Mozilla",
    "license": "MPL-2.0",
    "homepage": "https://github.com/mozilla/DeepSpeech/tree/v0.7.0-alpha.3#project-deepspeech",
    "files": [
       "README.md",
       "client.js",
       "index.js",
       "lib/*"
    ],
    "bugs": {
       "url": "https://github.com/mozilla/DeepSpeech/issues"
    },
    "repository"   :  {
       "type" : "git",
       "url"  : "git://github.com/mozilla/DeepSpeech.git"
    },
    "binary": {
        "module_name" : "deepspeech",
        "module_path" : "./lib/binding/v{version}/{platform}-{arch}/{node_abi}/",
        "remote_path" : "./v{version}/{configuration}/",
        "package_name": "{module_name}-v{version}-{node_abi}-{platform}-{arch}.tar.gz",
        "host"        : "https://community-tc.services.mozilla.com/api/index/v1/task/project.deepspeech.tensorflow.pip.v1.0.0-warpctc.arm/artifacts/public/"
    },
    "dependencies"  : {
      "node-pre-gyp": "0.14.x",
      "argparse": "1.0.x",
      "sox-stream": "2.0.x",
      "memory-stream": "0.0.3",
      "node-wav": "0.0.2"
    },
    "devDependencies": {
    },
    "scripts": {
        "test": "node index.js"
    }
}
npm install node-pre-gyp@0.14.x dts-gen@0.6.0
npm WARN deprecated mkdirp@0.5.4: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
+ dts-gen@0.6.0
+ node-pre-gyp@0.14.0
updated 2 packages and audited 188 packages in 3.472s
found 0 vulnerabilities

$(npm root)/.bin/dts-gen -module deepspeech -file index.d.ts
Invalid arguments: Cannot specify more than one output mode

Usage: dts-gen input [settings] [output]

Input Options:
  -m[odule] fs            The "fs" node module (must be installed)
  -i[dentifier] Math      The global variable "Math"
  -e[xpression] "new C()" The expression "new C()"
  -t[emplate] module      Name of a template. Templates are:
                          global-modifying-module, global-plugin, global, module-class, module-function, module-plugin, module

Settings:
  -n[ame] n               The name to emit when generating for an expression

Output Options:
  -f[ile] [filename.d.ts]  Write to a file (default)
  -d[t] [dirName]          Create a folder suitable for DefinitelyTyped
  -s[tdout]                Write to stdout
  -o[verwrite]             Allow overwriting files

Example: dts-gen -m fs --stdout
make: *** [Makefile:46: npm-pack] Error 1
make: Leaving directory '/home/alex/codaz/Mozilla/DeepSpeech/deepspeech-kdavis/native_client/javascript'
```

The doc seems to say we need to have `deepspeech` installed for `-module` to work.

Comment (lissyx) [reply native_client/javascript/Makefile]: @piraka9011 Do we really need to have the module installed ? This is going to be more tricky than expected if it's the case ...

Comment (piraka9011) [reply native_client/javascript/Makefile:46]: If it's not installed in the global namespace and not in the path, then `npx run dts-gen...` might work instead.

**Comment by lissyx** [comment]
Supersed by #2876

**Pull request closed without merging**